runtime/interface.go

```diff
@@ -102,138 +179,131 @@ type Interface interface {
 	) (bool, error)
 	// Hash returns the digest of hashing the given data with using the given hash algorithm
 	Hash(data []byte, hashAlgorithm string) ([]byte, error)
-	// GetStorageUsed gets storage used in bytes by the address at the moment of the function call.
-	GetStorageUsed(address Address) (value uint64, err error)
-	// GetStorageCapacity gets storage capacity in bytes on the address.
-	GetStorageCapacity(address Address) (value uint64, err error)
-}
-
-type HighLevelStorage interface {
-	Interface
-
-	// HighLevelStorageEnabled should return true
-	// if the functions of HighLevelStorage should be called,
-	// e.g. SetCadenceValue
-	HighLevelStorageEnabled() bool
-
-	// SetCadenceValue sets a value for the given key in the storage, owned by the given account.
-	SetCadenceValue(owner Address, key string, value cadence.Value) (err error)
 }
 
 type Metrics interface {
+	// ProgramParsed captures the time spent on parsing the program
 	ProgramParsed(location common.Location, duration time.Duration)
+	// ProgramChecked captures the time spent on checking the parsed program
 	ProgramChecked(location common.Location, duration time.Duration)
+	// ProgramInterpreted captures the time spent on interpreting the parsed and checked program
 	ProgramInterpreted(location common.Location, duration time.Duration)
+
+	// ValueEncoded captures the time spent on encoding a value
+	// TODO: maybe add type
```

Comment:
What "type" does the TODO refer to? The type of the encoded value?

Reply:
yeah something like that, more info on what was encoded or decoded. so we can have some info on what is costly right now on chain.